From 4c159040d59e176646fadd16b5d36c8fe35be92f Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Thu, 16 Feb 2023 12:06:44 -0500
Subject: [PATCH] tap-ctl: Default to read-only opening

OpenXT is changing to hash .vhds directly.  To do so, we need to only
open .vhds as read-only as read-write can modify the metadata even if
the contents do not change.

Change the default to read-only, the former -R option.  Add a new -W
option to use when read-write is desired.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 control/tap-ctl.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/control/tap-ctl.c
+++ b/control/tap-ctl.c
@@ -307,11 +307,11 @@ tap_cli_create(int argc, char **argv)
 	devname   = NULL;
 	secondary = NULL;
 	prt_minor = -1;
-	flags     = 0;
+	flags     = TAPDISK_MESSAGE_FLAG_RDONLY;
 	timeout   = 0;
 
 	optind = 0;
-	while ((c = getopt(argc, argv, "a:c:RDd:e:r2:st:C:h")) != -1) {
+	while ((c = getopt(argc, argv, "a:c:RWDd:e:r2:st:C:h")) != -1) {
 		switch (c) {
 		case 'a':
 			args = optarg;
@@ -326,6 +326,9 @@ tap_cli_create(int argc, char **argv)
 		case 'R':
 			flags |= TAPDISK_MESSAGE_FLAG_RDONLY;
 			break;
+		case 'W':
+			flags &= ~TAPDISK_MESSAGE_FLAG_RDONLY;
+			break;
 		case 'D':
 			flags |= TAPDISK_MESSAGE_FLAG_NO_O_DIRECT;
 			break;
